modules/market/src/main/java/com/opengamma/strata/market/param/ParameterizedDataCombiner.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.param;
+
+import com.google.common.base.Preconditions;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * Helper that can be used to combine two or more underlying instances of {@code ParameterizedData}.
+ * <p>
+ * This is used by implementations of {@link ParameterizedData} that are based on more
+ * than one underlying {@code ParameterizedData} instance.
+ * <p>
+ * This helper should be created in the constructor of the combined instance.
+ * In each of the five {@code ParameterizedData} methods of the combined instance,
+ * this helper should be invoked. See {@code DiscountFxForwardRates} for sample usage.
+ */
+public final class ParameterizedDataCombiner {
+
+  /**
+   * The underlying instances.
+   */
+  private final ParameterizedData[] underlyings;
+  /**
+   * The lookup array.
+   */
+  private final int[] lookup;
+  /**
+   * The count of parameters.
+   */
+  private final int paramCount;
+
+  /**
+   * Obtains an instance that can combine the specified underlying instances.
+   * 
+   * @param instances  the underlying instances to combine
+   * @return the combiner
+   */
+  public static ParameterizedDataCombiner of(ParameterizedData... instances) {
+    return new ParameterizedDataCombiner(instances);
+  }
+
+  //------------------------------------------------------------------------- 
+  // creates an instance
+  private ParameterizedDataCombiner(ParameterizedData[] underlyings) {
+    ArgChecker.notEmpty(underlyings, "underlyings");
+    int size = underlyings.length;
+    this.underlyings = underlyings;
+    int[] lookup = new int[size];
+    for (int i = 1; i < size; i++) {
+      lookup[i] = lookup[i - 1] + underlyings[i - 1].getParameterCount();
```

Comment:
Can we add something along these lines to the code standards in the wiki? I did a search there before making these comments, but your explanation makes sense. I'll merge it now.